crates/sncast/src/starknet_commands/account/import.rs

```diff
@@ -98,36 +94,55 @@ pub async fn add(
         _ => fetched_class_hash,
     };
 
-    let legacy = check_if_legacy_contract(class_hash, add.address, provider).await?;
+    let chain_id = get_chain_id(provider).await?;
+    if let (Some(salt), Some(class_hash)) = (import.salt, class_hash) {
+        let sncast_account_type = match import.account_type {
+            AccountType::Argent => SNCastAccountType::Argent,
+            AccountType::Braavos => SNCastAccountType::Braavos,
+            AccountType::Oz => SNCastAccountType::OpenZeppelin,
+        };
```

Comment:
In what case should it be possible for `class_hash` to be `None`? It should be just `Felt` and we shouldn't have `if let` it here.

Reply:
You're right, at this point in code we know that class hash is not `None`. I refactored class hash resolution logic, should be better now.